tests/helpers/periodPickerHelper-test.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * periodPickerHelper-test.js
+ * Created by Kwadwo Opoku-Debrah 12/07/18
+ */
+
+import * as periodPickerHelper from 'helpers/periodPickerHelper';
+
+const nativeDate = Date.now;
+
+const mockDate = (date) => {
+    // mock the current date
+    const month = parseInt(date.substring(5, 7), 10) - 1; // month is zero-indexed
+    const day = parseInt(date.substring(8));
+    const mock = new Date(date.substring(0, 4), month, day);
+    Date.now = () => mock;
+};
+
+afterAll(() => {
+    // restore the original, native date function
+    Date.now = nativeDate;
+});
+
+describe('periodPickerHelper', () => {
+    describe('handlePotentialStrings', () => {
+        it('should convert its input to an integer if it is a string', () => {
+            const output = periodPickerHelper.handlePotentialStrings('12');
+            expect(output).toEqual(12);
+        });
+        it('should return its input otherwise', () => {
+            const output = periodPickerHelper.handlePotentialStrings(12);
+            expect(output).toEqual(12);
+        });
+    });
+
+    describe('mostRecentPeriod', () => {
+        it('should return the most recently completed period', () => {
+            mockDate('1912-06-01');
+            const output = periodPickerHelper.mostRecentPeriod();
+            expect(output.period).toEqual(6);
+        });
+        it('should return the fiscal year of the most recently completed period', () => {
+            mockDate('1912-06-01');
+            const output = periodPickerHelper.mostRecentPeriod();
+            expect(output.year).toEqual(1912);
+        });
+        it('if GTAS data is not available, it should return the period before that', () => {
+            mockDate('1912-04-01');
+            const output = periodPickerHelper.mostRecentPeriod();
+            expect(output).toEqual({
+                period: 3,
+                year: 1912
+            });
+        });
+        it('if GTAS data is not available for period 1, then it should return period 12 of the previous fiscal year', () => {
```

Comment:
Period one is not directly selectable, so this should check for a date in period 2 when GTAS data is not yet available for the current FY (e.g. today on Dec 17, we are in FY2019 Period 2, but it should return FY2018 Period 12)